File: admin-service/src/main/java/com/acadprobot/admin/service/SuperAdminService.java
package com.acadprobot.admin.service;

import com.acadprobot.admin.model.Admin;
import com.acadprobot.admin.model.AdminChatbotRequest;
import com.acadprobot.admin.repository.AdminChatbotRequestRepository;
import com.acadprobot.admin.repository.AdminRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import com.acadprobot.admin.repository.ChatbotRepository;
import com.acadprobot.admin.model.Chatbots;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.*;

@Service
public class SuperAdminService {
    @Autowired
    private ChatbotRepository chatbotRepository;

    @Autowired
    private AdminRepository adminRepository;

    @Autowired
    private AdminChatbotRequestRepository adminChatbotRequestRepository;

    public Chatbots createChatbot(String chatbotName, String adminEmail){
        Admin admin = adminRepository.findByEmail(adminEmail)
                .orElseThrow(()->new RuntimeException("Admin not found: " + adminEmail));

        Chatbots newChatbot = new Chatbots();
        newChatbot.setAdmin(admin);
        newChatbot.setName(chatbotName);

        return chatbotRepository.save(newChatbot);
    }

    public List<Chatbots> getAllChatbots(){
        return chatbotRepository.findAll();
    }

    public ResponseEntity<?> deleteChatbot(@PathVariable UUID id){
        chatbotRepository.deleteById(id);
        return ResponseEntity.ok("Chatbot deleted successfully");
    }

    // admin chatbot request handling
    public AdminChatbotRequest createAdminChatbotRequest(
            String email,
            String fullname,
            String title,
            String chatbot_name,
            String departmentProgram,
            String purpose
    ){
        AdminChatbotRequest adminChatbotRequest = new AdminChatbotRequest();
        adminChatbotRequest.setEmail(email);
        adminChatbotRequest.setFullname(fullname);
        adminChatbotRequest.setTitle(title);
        adminChatbotRequest.setChatbot_name(chatbot_name);
        adminChatbotRequest.setDepartment_program(departmentProgram);
        adminChatbotRequest.setPurpose(purpose);

        return adminChatbotRequestRepository.save(adminChatbotRequest);
    }

    public List<AdminChatbotRequest> getAllRequestByStatus(String status){
        return adminChatbotRequestRepository.findByStatus(status);
    }

    public String approveRequest(UUID id){
        Optional<AdminChatbotRequest> requestOpt = adminChatbotRequestRepository.findById(id);
        if(requestOpt.isEmpty()){
            return "Request not found" + id;
        }
        AdminChatbotRequest adminChatbotRequest = requestOpt.get();
        adminChatbotRequest.setStatus("approved");

        adminChatbotRequestRepository.save(adminChatbotRequest);

        return "Request approved for " + adminChatbotRequest.getEmail();
    }

    public String rejectRequest(UUID id){
        Optional<AdminChatbotRequest> requestOpt = adminChatbotRequestRepository.findById(id);
        if(requestOpt.isEmpty()){
            return "Request not found" + id;
        }
        AdminChatbotRequest adminChatbotRequest = requestOpt.get();
        adminChatbotRequest.setStatus("rejected");

        adminChatbotRequestRepository.save(adminChatbotRequest);

        return "Request rejected for " + adminChatbotRequest.getEmail();
    }

}

File: frontend/src/services/superadminService.js
import springapi from './springapi'

const SPRING_API = "/superadmin";

export const createChatbot = async (chatbotName, adminEmail) => {
    const response = await springapi.post(`${SPRING_API}/createchatbot`, { chatbotName, adminEmail });
    return response.data;
}

export const getAllChatbots = async () => {
    const response = await springapi.get(`${SPRING_API}/chatbots`);
    return response.data;
}

export const deleteChatbot = async (id) => {
    return await springapi.delete(`${SPRING_API}/chatbots/${id}`)
}

export const requestAdminChatbot = async (
    email,
    fullname,
    title,
    chatbot_name,
    department_program,
    purpose
) => {
    const response = await springapi.post(`${SPRING_API}/requestadmin`, {
        email,
        fullname,
        title,
        chatbot_name,
        department_program,
        purpose
    });
    return response.data;
}


export const getAllRequest = async(status) =>{
    const response = await springapi.get(`${SPRING_API}/requestadmin`,{
        params: {status}
    });
    return response.data;
}

export const approveRequest = async(request_id) =>{
    const response = await springapi.post(`${SPRING_API}/approverequest`, null,{
        params: {request_id}
    })
    return response.data;
}

export const rejectRequest = async(request_id) =>{
    const response = await springapi.post(`${SPRING_API}/rejectrequest`, null, {
        params: {request_id}
    })
    return response.data;
}

export const downloadReport = async() =>{
    const response = await springapi.get(`${SPRING_API}/downloadreport`, {
    responseType: 'blob',
  })
    return response.data;
}